internal/controller/workspace/workspace.go

```diff
@@ -322,7 +323,23 @@ func (c *external) Delete(ctx context.Context, mg resource.Managed) error {
 		return errors.Wrap(err, errOptions)
 	}
 
-	return errors.Wrap(c.tf.Destroy(ctx, o...), errDestroy)
+	if err := c.tf.Destroy(ctx, o...); err != nil {
+		return errors.Wrap(err, errDestroy)
+	}
+	labels := map[string]string{"tfstate": "true", "tfstateWorkspace": cr.Name}
+	sl := &corev1.SecretList{}
+	_ = c.kube.List(ctx, sl, client.MatchingLabels(labels))
+	for s := range sl.Items {
+		sec := sl.Items[s]
+		_ = c.kube.Delete(ctx, &sec)
+	}
+	ll := &coordv1.LeaseList{}
+	_ = c.kube.List(ctx, ll, client.MatchingLabels(labels))
+	for l := range ll.Items {
+		ls := ll.Items[l]
+		_ = c.kube.Delete(ctx, &ls)
```

Comment:
We need to catch errors during this interaction with the kubernetes API

```
11:42:40 [ .. ] golangci-lint
internal/controller/workspace/workspace.go:331:2: Error return value of `c.kube.List` is not checked (errcheck)
	_ = c.kube.List(ctx, sl, client.MatchingLabels(labels))
	^
internal/controller/workspace/workspace.go:334:3: Error return value of `c.kube.Delete` is not checked (errcheck)
		_ = c.kube.Delete(ctx, &sec)
		^
internal/controller/workspace/workspace.go:337:2: Error return value of `c.kube.List` is not checked (errcheck)
	_ = c.kube.List(ctx, ll, client.MatchingLabels(labels))
	^
internal/controller/workspace/workspace.go:340:3: Error return value of `c.kube.Delete` is not checked (errcheck)
		_ = c.kube.Delete(ctx, &ls)
		^
11:42:49 [FAIL]
```

I've created #53 to enable automated check for that(slightly surprised that it was disabled :) )